### PR TITLE
Add changelog and changelog verifier

### PR DESCRIPTION
**Pull request by harshavamsi**
Signed-off-by: Harsha Vamsi Kalluri <harshavamsi096@gmail.com>

### Description
Introduces changelog and changelog verifier.

### Issues Resolved
_List any issues this PR will resolve, e.g. Closes [...]._ 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by Yury-Fridlyand — APPROVED** [review]

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -80,6 +82,22 @@ Signed-off-by: Jane Smith <jane.smith@email.com>
 
 You may type this line on your own when writing your commit messages. However, if your user.name and user.email are set in your git configs, you can use `-s` or `– – signoff` to add the `Signed-off-by` line to the end of the commit message.
 
+## Changelog
+
+OpenSearch-Net maintains version specific changelog by enforcing a change to the ongoing [CHANGELOG](CHANGELOG.md) file adhering to the [Keep A Changelog](https://keepachangelog.com/en/1.0.0/) format.
```
Comment (Xtansia) [inline CONTRIBUTING.md:87]:
```suggestion
OpenSearch-Net maintains a version specific changelog by enforcing a change to the ongoing [CHANGELOG](CHANGELOG.md) file adhering to the [Keep a Changelog](https://keepachangelog.com/en/1.0.0/) format.
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -80,6 +82,22 @@ Signed-off-by: Jane Smith <jane.smith@email.com>
...
+
+Briefly, the changes are curated by version, with the changes to the main branch added chronologically to `Unreleased` version. Further, each version has corresponding sections which list out the category of the change - `Added`, `Changed`, `Deprecated`, `Removed`, `Fixed`, `Security`.
```
Comment (Xtansia) [inline CONTRIBUTING.md:89]:
```suggestion
Briefly, the changes are curated by version, with the changes to the main branch added chronologically to the `Unreleased` version. Further, each version has corresponding sections which list out the category of the change - `Added`, `Changed`, `Deprecated`, `Removed`, `Fixed`, `Security`.
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -80,6 +82,22 @@ Signed-off-by: Jane Smith <jane.smith@email.com>
...
+
+
+### How to add my changes to [CHANGELOG](CHANGELOG.md)?
```
Comment (Xtansia) [inline CONTRIBUTING.md:92]: If you want to frame this as a question from the contributors perspective I'd go with:
```suggestion
### How do I add my changes to the [CHANGELOG](CHANGELOG.md)?
```

Comment (Xtansia) [reply CONTRIBUTING.md:92]: Otherwise if you want it as a statement of us telling the contributor it'd be more like:
```suggestion
### How to add your changes to the [CHANGELOG](CHANGELOG.md)
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -80,6 +82,22 @@ Signed-off-by: Jane Smith <jane.smith@email.com>
...
+
+As a contributor, you must ensure that every pull request has the changes listed out within the corresponding version and appropriate section of [CHANGELOG](CHANGELOG.md) file.
+
+Adding in the change is two step process -
```
Comment (Xtansia) [inline CONTRIBUTING.md:96]:
```suggestion
As a contributor, you must ensure that every pull request has its changes listed out within the corresponding version and appropriate section of the [CHANGELOG](CHANGELOG.md) file.

Adding in the change is a two step process -
```

File: .github/pull_request_template.md [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+### Issues Resolved
+[List any issues this PR will resolve]
+
+### Check List
+- [ ] New functionality includes testing.
+  - [ ] All tests pass
+- [ ] New functionality has been documented.
+  - [ ] New functionality has comments added
+- [ ] Commits are signed per the DCO using --signoff
+- [ ] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](https://github.com/opensearch-project/opensearch-net/blob/main/CONTRIBUTING.md#changelog))
```
Comment (Xtansia) [inline .github/pull_request_template.md:13]:
```suggestion
- [ ] Commit changes are listed out in the CHANGELOG.md file (See: [Changelog](https://github.com/opensearch-project/opensearch-net/blob/main/CONTRIBUTING.md#changelog))
```

File: .github/pull_request_template.md [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+
+By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
+For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/opensearch-net/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
```
Comment (Xtansia) [inline .github/pull_request_template.md:16]:
```suggestion
For more information on following the Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/opensearch-net/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
```

**Review by Xtansia — CHANGES_REQUESTED** [review]
Overall looks really good. Just some pretty minor grammar and phrasing suggestions from me

**Comment by dblock** [comment]
Good catch @Xtansia. I have more in https://github.com/opensearch-project/opensearch-go/pull/172/.

**Review by dblock — APPROVED** [review]

**Review by Yury-Fridlyand — APPROVED** [review]

**Review by VachaShah — APPROVED** [review]

**Comment by dblock** [comment]
@Xtansia you good with this?

File: MAINTAINERS.md [hunk 7]
```diff
@@ -1,147 +1,17 @@
...
-  - [Interest](#interest)
-  - [Addition](#addition)
-- [Removing a Maintainer](#removing-a-maintainer)
-  - [Moving On](#moving-on)
-  - [Inactivity](#inactivity)
-  - [Negative Impact on the Project](#negative-impact-on-the-project)
 ## Overview
 
-This document explains who the maintainers are (see below), what they do in this repo, and how they should be doing it. If you're interested in contributing, see [CONTRIBUTING](CONTRIBUTING.md).
+[This document](https://github.com/opensearch-project/.github/blob/main/MAINTAINERS.md#maintainer-responsibilities) explains who the maintainers are (see below), what they do in this repo, and how they should be doing it. If you're interested in contributing, see [CONTRIBUTING](CONTRIBUTING.md).
```
Comment (Xtansia) [inline MAINTAINERS.md:5]: Saying "This document" linking to another document, while explaining the contents of the current document and the linked one is somewhat confusing. I think it'd be better to rephrase this to something like `This document lists the current maintainers of this repository. An explanation of what they do and their responsibilities can be found in the [central opensearch project MAINTAINERS file]`

**Review by Xtansia — COMMENTED** [review]
LGTM, just one comment, happy for it to be merged regardless though

File: MAINTAINERS.md (hunk 7 above)
Comment (dblock) [reply MAINTAINERS.md:5]: I hit merge - this is a valid comment though, we have this language all over the place. Something to add to a template ...
